src/index.js

```diff
@@ -1,7 +1,295 @@
+          )
+        }
+    } else {
+      return (
+        <div id="info">
+          <p>Latitude: ?</p>
+          <p>Longitude: ?</p>
+          <p>County: ?</p>
+          <p>Municipality: ?</p>
+          <p>Score: {this.props.score}</p>
+        </div>
+      )
+    }
+  }
+}
+
+class Controls extends React.Component {
+  render () {
+    return (
+      <div id="controls">
+        <h2 style={{gridArea: "title"}}>Controls:</h2>
+        <button id="up" style={{gridArea: "up"}}>↑</button>
+        <button id="left" style={{gridArea: "left"}}>←</button>
+        <button id="right" style={{gridArea: "right"}}>→</button>
+        <button id="down" style={{gridArea: "down"}}>↓</button>
+      </div>
+    )
+  }
+}
 
 ReactDOM.render(
-  <h1>Hello World!</h1>,
+  <Page />,
   document.getElementById('root')
-)
\ No newline at end of file
+)
+
+function randomNumber(min, max) {
+  let range = max - min; 
+  return min + (Math.random() * range);
+}
+
+function findVermontLocation () {
+  let location = randomInRectangle();
+  let i = 0;
+  while (!withinVermont(location) && i<100) {
+    console.log(location.lng + ', ' + location.lat + " is outside of vermont, finding a new location...")
+    location = randomInRectangle();
+    i++;
+  }
+  if (!withinVermont(location)) {
+    console.log("I failed to find a location within Vermont's borders :(")
+  } else {
+    console.log("After " + (i + 1) + " try(ies)...")
+    return location;
+  }
+}
+
+function randomInRectangle() { //"rectangle" meaning a rectangle that encloses all of VT
+  let result = {
+    lng: randomNumber(-73.41592488911837, -71.51022535353107),
+    lat: randomNumber(42.730315121762715, 45.007561302382754)
+  }
+  return result;
+}
+
+function withinVermont(location) {
+  let array = leafletPip.pointInLayer([location.lng, location.lat], L.geoJSON(borderData));
+  if (array.length === 0) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+function findLocationInfo(object, location) {
+  let areaNames = []; //e.g. street, county, country
+  let municipalityTypeIndex = null;
+  for (let item in object["address"]) {
+    areaNames.push(item);
+  }
+  for (let i = 0; i < areaNames.length - 1; i++) {
+    if (areaNames[i]==="county") { //town, village, city, hamlet etc. always 1 before county
+      municipalityTypeIndex = i - 1;
+    }
+  }
+  location["municipalityType"] = areaNames[municipalityTypeIndex]; //e.g. city, town, hamlet, village
+  location["municipalityName"] = object["address"][location["municipalityType"]] //e.g. burlington, stowe, rutland
+  location["state"] = object["address"]["state"] //double check that we're in vermont
+  location["county"] = object["address"]["county"]
+  console.log("I have selected a location with longitude " + location.lng + " and latitude " + location.lat +
+    " in the " + location.municipalityType + " of " + location.municipalityName + ", " + location.county + ", " + location.state + ".");
+  return location;
+}
```

Comment:
Make sure you document the purpose and practicality of your functions with comments. It will improve your project's readability if you talk about how and why certain functions work.